python/ray/train/torch/config.py

```diff
@@ -16,6 +16,19 @@
 logger = logging.getLogger(__name__)
 
 
+class TorchConfigContextManager:
```

Comment:
Oh I think it's fine because each backend returns their own context manager, which can be either function-based or class-based. Also we can subclass the existing context manager like:

```python
class OuterContextManager(InnerContextManager):
    def __enter__(self):
        super().__enter__()
        ...
        return self

    def __exit__(self, exc_type, exc_val, exc_tb):
        super().__exit__(exc_type, exc_val, exc_tb)
        ...
        return False
```

Reply:
I think it's fine since you can return either function-based or class-based context manager.

To use the existing context manager, we can subclass it and make it nested:

```python
class InnerContextManager:
    def __enter__(self):
        print("Entering InnerContextManager")
        return self

    def __exit__(self, exc_type, exc_val, exc_tb):
        print("Exiting InnerContextManager")
        return False

class OuterContextManager(InnerContextManager):
    def __enter__(self):
        print("Entering OuterContextManager")
        super().__enter__()
        return self

    def __exit__(self, exc_type, exc_val, exc_tb):
        super().__exit__(exc_type, exc_val, exc_tb)
        print("Exiting OuterContextManager")
        return False 

```